### PR TITLE
allow selecting a preferred source address on exported routes

### PR DESCRIPTION
**Pull request by christf**
This will allow exporting routes having the RTA_PREFSRC field set and adds the command line option -P to set this very address.  This is interesting for multi-homed nodes to avoid leaking addresses into the mesh that do not belong.

In the routing table these routes then look like this:

```
fec0::ae6:89ff:feba:e632 via fe80::a800:ff:fed5:4ced dev ens6 proto babel src 2a06:8187:fbab:1::9000:2 metric 106 onlink pref medium
```

### CODE REVIEW

File: babeld.man [hunk 1]
```diff
@@ -91,6 +90,10 @@ explicitly overridden in the configuration file.
 Do not perform split-horizon processing on wired interfaces.
 Split-horizon is not performed on wireless interfaces.
 .TP
+.BI \-P " preferred-ipv6-source-address"
+This specifies the ipv6 address which will be added to kernel routes as
+preferred src address.
```
Comment (mweinelt) [inline babeld.man]: Proposal:

> Specify the IPv6 address to be set as the preferred source address for routes exported to the kernel.

Comment (christf) [reply babeld.man]: yes, that is better.

**Comment by jech** [comment]
I don't think it should be a command-line option, I think it should be something in the config file.  I can see two ways:

1. a global option (see `parse_option` in `configuration.c`), in which case it will apply to all routes;
2. a per-interface option (see `parse_ifconf`), in which case it would apply to all routes through a given interface;
3. a filter option (see `parse_filter`), in which case it will apply to selected routes.

(2) feels like the reasonable compromise between complexity and flexibility.  Do you want to summarise this on the list?

**Comment by christf** [comment]
as discussed via Mail, picking option (2) for now and rebasing this on top of the unicast branch.

**Comment by jech** [comment]
I'm confused, the PREFSRC branch still uses a command-line option.  Which branch do you want me to look at?

**Comment by christf** [comment]
On Mon, Jul 16, 2018 at 04:07:40AM -0700, Juliusz Chroboczek wrote:
>I'm confused, the PREFSRC branch still uses a command-line option.  Which branch do you want me to look at?
Sorry, I have not reworked this yet. I want to 
* make this a configuration via config file & socket
* make this interface specific
  
This hasn't happened yet, just a rebase so far.
-- 
()  ascii ribbon campaign - against html e-mail
/\  against proprietary attachments

**Comment by christf** [comment]
The patch has now been reworked to allow for configuration via the config file in a line similare to this:
interface babel-vpn-1374 type tunnel link-quality true update-interval 300 pref-src aaaa:bbbb:cccc:dddd:eeee::fffff

**Comment by christf** [comment]
... closing this PR in favor of #17

**Pull request closed without merging**